From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sleigh-InSPECtor <sleighinspector@outlook.com>
Date: Thu, 23 May 2024 13:46:50 +0930
Subject: [PATCH] 6564: AArch32: (Thumb32) ldrsh.w & ldrsb.w (pc-relative) had
 double memory load

AArch32: (Thumb32) fixed ldrsh.w & ldrsb.w (pc-relative)

* double memory load bug
---
 .../Processors/ARM/data/languages/ARMTHUMBinstructions.sinc   | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
index 8ac0192941..80594011ba 100644
--- a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
@@ -2397,7 +2397,7 @@ define pcodeop ExclusiveAccess;
 :ldrsb^ItCond^".w"  Rt1215,PcrelOffset12         is TMode=1 & ItCond & (op8=0xf9 & thc0506=0 & thc0404=1 & sop0003=15; Rt1215) & PcrelOffset12
 {
    build ItCond;
-   tmp:1 = *PcrelOffset12;
+   tmp:1 = PcrelOffset12:1;
    Rt1215 = sext(tmp);
 }
 
@@ -2438,7 +2438,7 @@ define pcodeop ExclusiveAccess;
 {
    build ItCond;
    build PcrelOffset12;
-   tmp:2 = *PcrelOffset12;
+   tmp:2 = PcrelOffset12:2;
    Rt1215 = sext(tmp);
 }
 
-- 
2.45.1

